flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/utils/StreamingWithStateTestBase.scala

```diff
@@ -68,8 +69,8 @@ class StreamingWithStateTestBase(state: StateBackendMode) extends StreamingTestB
     super.before()
     // set state backend
 
-    // subfolder are managed here because the tests could fail during cleanup when concurrently executed (see FLINK-33820)
-    baseCheckpointPath = TempDirUtils.newFolder(tempFolder)
+    val baseCheckpointPath = Files.createTempDirectory(getClass.getCanonicalName)
+    Files.deleteIfExists(baseCheckpointPath);
     state match {
```

Comment:
Thanks @snuyanzin for this hard work.

But I am a little confused, why is `baseCheckpointPath` defined as a stack variable, so does the object variable `baseCheckpointPath: File = _` have no effect anymore?

I suspect that there are still some problems about reuse the `tempFolder` by `TempDirUtils.newFolder(tempFolder)` instead of directly using `Files.createTempDirectory("junit")`.
I think the cleanup of `tempFolder` may still fail when `FileUtils.deleteDirectory(baseCheckpointPath)` met `DirectoryNotEmptyException` because it is possible that there are some threads still writing to the folder.

Reply:
Since `baseCheckpointPath` is used only in this method I do not see any reason to have it as a field, so I just moved it completely to the method

Reply:
Sorry, if I was wrong, please correct me. 
I still don't quite understand why we need to delete the folder we just created immediately.

Reply:
suddenly.. :see_no_evil: so far I thought it was `baseCheckpointPath.toFile.deleteOnExit();`, thanks !
now changed it